### PR TITLE
Added cypress-test for code-highlighting

### PR DESCRIPTION
**Pull request by anishagg17**
**Description**
Added cypress-test for code-highlighting

**Screenshots**

Screenshot 2021-08-02 at 6 57 57 PM

Screenshot 2021-08-02 at 6 57 44 PM

**Checks**
- [ ] The new code matches the existing patterns and styles.
- [ ] The tests pass with `yarn test`.
- [ ] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [ ] The relevant examples still work. (Run examples with `yarn start`.)
- [ ] You've [added a changeset](https://github.com/atlassian/changesets/blob/master/docs/adding-a-changeset.md) if changing functionality. (Add one with `yarn changeset add`.)

### CODE REVIEW

File: cypress/integration/examples/codeHighlighting.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+describe('code highlighting', () => {
```
Comment (timbuckley) [inline cypress/integration/examples/codeHighlighting.ts:1]: The file name should use the hyphenated convention used elsewhere, so `examples/code-highlighting.ts`

Comment (anishagg17) [reply cypress/integration/examples/codeHighlighting.ts]: Resolved in some other examples too that I noticed have `fileName` different from that in `site/examples` folder 🤗

**Review by timbuckley — APPROVED** [review]

File: cypress/integration/examples/code-highlighting.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    cy.get(slateEditor)
+      .type('{movetostart}')
+      .type(JSCode) // Type JavaScript code
+
+    cy.get(slateEditor)
+      .find('span')
+      .eq(0)
+      .find(leafNode)
+      .eq(0)
+      .should('contain', 'const')
```
Comment (dylans) [inline cypress/integration/examples/code-highlighting.ts:32]: This test is causing several new PRs to fail, specifically #3888 and #4037:

```bash
  code highlighting
    ✓ highlights HTML tags (16743ms)
event - build page: /examples/[example]
wait  - compiling...
    1) highlights javascript syntax


  1 passing (39s)
  1 failing

  1) code highlighting
       highlights javascript syntax:
     AssertionError: Timed out retrying after 4000ms: expected '<span.css-14qid8c>' to contain 'const'
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/examples/code-highlighting.ts:124:14)
```

Is it new code that's broken or is does this test not appear to be accurate?

Comment (anishagg17) [reply cypress/integration/examples/code-highlighting.ts:32]: I think we should increase the timout, the test is new one but is fine. Might be unable to render the changes within the current timeframe.

**Pull request merged**